howso/client/base.py

```diff
@@ -3036,6 +3036,316 @@ def _react_series(self, trainee_id: str, params: dict):
+            series whose stationary feature values are to be predicted
+        series_id_features : list of str, optional
+            List of feature names corresponding to the values in each row of
+            ``series_id_values``. This value is ignored if ``series_id_values``
+            is not specified. If specified, all series ID features should be
+            contained within the given list.
+        series_id_values : list of list of object, optional
+            2d list of ID feature values. Each sublist should specify ID
+            feature values that can uniquely identify the cases making up a
+            single series.
+        use_case_weights : bool, optional
+            If True, then the Trainee will use case weights identified by the
+            name given in ``weight_feature``. If False, case weights will not
+            be used. If unspecified, case weights will be used if the Trainee
+            has them.
+        use_derived_ts_features : bool, default True
+            If True, then time-series features derived from features specified
+            as contexts will additionally be added as context features.
+        use_regional_residuals : bool, default True
+            If False, global residuals will be used in generative predictions.
+            If True, regional residuals will be computed and used instead. This
+            may increase runtime noticeable.
+        weight_feature : str, optional
+            The name of the weight feature to be used. Should be used in
+            combination with ``use_case_weights``.
+
+        Returns
+        -------
+        Reaction
+            A MutableMapping (dict-like) with these keys -> values:
+                action -> pandas.DataFrame
+                    A DataFrame of action values.
+                details -> dict or list
+                    A dict containing details.
+
+        Raises
+        ------
+        ValueError
+            If `action_features` is not a list of strings.
+            If `context_features` is not a list of strings.
+            If `series_context_features` is not a list of strings.
+            If `series_id_features` is not a list of strings.
+
+            If both `series_id_values` and `series_context_values` are
+            specified.
+        """
+        trainee_id = self._resolve_trainee(trainee_id).id
+        feature_attributes = self.resolve_feature_attributes(trainee_id)
+        util.validate_list_shape(action_features, 1, "action_features", "str")
+        util.validate_list_shape(context_features, 1, "context_features", "str")
+        util.validate_list_shape(series_context_features, 1, "series_context_features", "str")
+        util.validate_list_shape(series_id_features, 1, "series_id_features", "str")
+
+        if (series_id_values is not None and series_context_values is not None) or \
+           (series_id_values is None and series_context_values is None):
+            raise ValueError((
```

Comment:
Prefer parenthesis over escaping the line ending
```suggestion
        if (
            (series_id_values is not None and series_context_values is not None) or
            (series_id_values is None and series_context_values is None)
        ):
```